test/test_helper.rb

```diff
@@ -39,9 +39,12 @@ class ActiveSupport::TestCase
   strategies = { active_record: :transaction,
                  mongoid: :truncation }
   DatabaseCleaner.strategy = strategies[DEVISE_TOKEN_AUTH_ORM]
+  DatabaseCleaner.clean_with(:truncation)
```

Comment:
why this change?

Reply:
It was an attempt to combat the mysterious database constraint violations with duplicate emails while testing. 

That validation problem I don't think is related to the changes on this branch, but I'm still trying to resolve it. I just made a change to ensure unique emails, so hopefully that fixes it